### PR TITLE
update charts os badges, remove filtering

### PR DESCRIPTION
**Pull request by mantis-toboggan-md**
#5137 
#4983 
This PR updates the os badges on charts to better distinguish between charts that deploy on windows nodes, and charts that just wont break if there are windows nodes present in the cluster (they have taints to avoid scheduling on those nodes). It also applies these badges to the Cluster Tools page. 
In this PR:
- OS badges are now applied to non-Rancher charts as well
- all charts deploy on Linux so there are no Linux badges anymore, instead there is a user-dismissable warning in the app marketplace, chart overview, chart install page, and cluster tools page if legacy tools are not enabled ('legacy' feature flag)
- charts with the `deploys-on=linux,windows` have a badge indicating they deploy on Windows - currently this is only Logging and Monitoring
- charts without a `permits-os` annotation or whose `permits-os` annotation does not include Windows have a badge on overview and install pages indicating that they don't work with Windows clusters - currently this is all partner charts
- if installing a version of a chart that doesn't support Windows, a badge indicating this will appear in the wizard

Screen Shot 2022-03-11 at 1 27 09 PM
Screen Shot 2022-03-11 at 1 15 27 PM
Screen Shot 2022-03-11 at 1 15 58 PM
Screen Shot 2022-03-11 at 1 16 23 PM
Screen Shot 2022-03-11 at 1 40 28 PM
Screen Shot 2022-03-11 at 1 38 56 PM

### CODE REVIEW

**Comment by mantis-toboggan-md** [comment]
Updated to say 'Linux only' instead of 'Incompatible with Windows'
Screen Shot 2022-03-11 at 2 38 04 PM

**Review by neillsom — COMMENTED** [review]
LG TM

File: pages/c/_cluster/apps/charts/index.vue [hunk 1]
```diff
@@ -256,16 +257,14 @@ export default {
...
       const compatibleVersions = compatibleVersionsFor(chart, OSs, showPrerelease);
+      const versions = chart.versions;
 
       if (compatibleVersions.length > 0) {
         version = compatibleVersions[0].version;
       } else {
-        console.warn(`Cannot select chart '${ chart.chartName }' (no compatible version available for '${ OSs.length > 1 ? `${ OSs[0] } and ${ OSs[1] }` : OSs[0] }')`); // eslint-disable-line no-console
-
-        return;
+        version = versions[0].version;
```
Comment (rak-phillip) [inline pages/c/_cluster/apps/charts/index.vue:267]: Is it ever possible for versions to be null or undefined? My concern is that we can throw an unhandled error when trying to access `version`.

**Review by rak-phillip — APPROVED** [review]
LGTM as well. Just one small question that I had.

File: pages/c/_cluster/apps/charts/index.vue (hunk 1 above)
Comment (mantis-toboggan-md) [reply pages/c/_cluster/apps/charts/index.vue:267]: That's a reasonable concern, but chart versions are defined with empty array [here](https://github.com/rancher/dashboard/blob/master/store/catalog.js#L512) then filled so this should be okay.

Comment (rak-phillip) [reply pages/c/_cluster/apps/charts/index.vue:267]: Sounds good, thanks for the head up 😁

**Pull request merged**